### ISSUE TITLE
OCaml 4.07.1 fails to install

### ISSUE THREAD

**Issue by lpil** [vic/asdf-ocaml#3]
```
louispilfold ~ $ asdf install ocaml 4.07.1
Checking for available remotes: rsync and local, git.
  - you won't be able to use mercurial repositories unless you install the hg
    command on your system.
  - you won't be able to use darcs repositories unless you install the darcs
    command on your system.

[WARNING] Recommended external solver aspcud not found.

=-=- Fetching repository information =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[default] synchronized from https://opam.ocaml.org
[NOTE] The repository 'default' will be *permanently* redirected to
       https://opam.ocaml.org/1.2.2 (opam-version < "2.0~")
[default] synchronized from https://opam.ocaml.org/1.2.2
Cannot find /home/louispilfold/.asdf/installs/ocaml/4.07.1/compilers/4.07.1/4.07.1/4.07.1.comp: 4.07.1 is not a valid compiler name.
[ERROR] Initialisation failed
OpamGlobals.Exit(1)
mv: cannot move '/tmp/ocaml_onH7qnB/opam' to '/home/louispilfold/.asdf/installs/ocaml/4.07.1/4.07.1/bin/': No such file or directory
```

4.07.0 does not have the same problem

**Comment by DanCouper** [comment]
Same issue, note this seems to affect anything > 4.07.0 (both of the 4.08 beta releases fail with the exact same message).

**Comment by MaciekTalaska** [comment]
@DanCouper @lpil 

I have also bumped on the save issue. It seems that very old version of OPAM was to blame.
I have forked this plugin, and created a fix for that, and issued a PR: https://github.com/vic/asdf-ocaml/pull/4

I am not sure why it fails under OSX (and I don't use one to check). 
If you want to check what I have done, just replace the `install` script inside `.asdf/plugins/ocaml/bin` to the one from my repo and check if that works for you

I hope that the original author is going to merge this soon! :)
